lib/backend/k8s/syncer.go

```diff
@@ -465,9 +502,8 @@ func (syn *kubeSyncer) performSnapshotDeletes(exists map[string]bool) {
 // a mapping of model.Key objects representing the objects which exist in the datastore, and
 // populates the provided resourceVersions with the latest k8s resource version
 // for each.
-func (syn *kubeSyncer) performSnapshot() ([]model.KVPair, map[string]bool, resourceVersions) {
+func (syn *kubeSyncer) performSnapshot(versions *resourceVersions) ([]model.KVPair, map[string]bool) {
```

Comment:
It's because of variable declaration scope changes.

There is this decl at the top of readFromKubernetesAPI:
```
	latestVersions := resourceVersions{}
```

The old performSnapshot call,
```
			snap, existingKeys, latestVersions := syn.performSnapshot()
```
used to _shadow_ that top-level decl.  But it didn't matter, because the other places where we _used_ latestVersions were within this same scope (i.e. `if needsResync ...`).

But now the other using places (i.e. the watch setup code) are _outside_ this scope.  So it is important that we get rid of the shadowing and ensure that everyone is playing with the same, top-level, latestVersions.